[WTF-2149] Update existing documentation for local variables

LEGIO-SEXTA-FERRATA opened the pull request:
**WARNING**: This PR is based on [this one](https://github.com/mendix/docs/pull/9130) and therefore should not be merged before it does. This branch must be rebased on master after the other is merged.

weirdwater committed:
Add documentation on the use of variables on pages and snippets

weirdwater committed:
Team feedback

weirdwater committed:
Mention that unsupported widgets are displayed in the modeler

weirdwater committed:
Update documentation regarding selectors and variables

weirdwater committed:
Apply feedback

LEGIO-SEXTA-FERRATA committed:
Update existing parts that are relevant to the local variables

alihcsumer commented on content/en/docs/refguide/modeling/pages/data-widgets/data-sources/_index.md:
```diff
@@ -16,8 +16,8 @@ This document describes data sources of data containers.
 * [Context](/refguide/context-source/) – the data view gets its object from the context: either from a page parameter or a surrounding data container.
-* [Microflow](/refguide/microflow-source/) – the data view object is determined by the result of calling the selected microflow. The microflow can take objects in the context as parameter and needs to return a single object.
-* [Nanoflow](/refguide/nanoflow-source/) – objects retrieved are determined by the result of calling the selected nanoflow. The nanoflow can take objects in the context as a parameter and needs to return a single object. 
+* [Microflow](/refguide/microflow-source/) – the data view object is determined by the result of calling the selected microflow. The microflow can take variables or objects in the context as parameter and needs to return a single object.
```
'Primitives or objects' sounds more appropriate than 'variables or objects.'

alihcsumer replied:
✨

alihcsumer commented on content/en/docs/refguide/modeling/pages/data-widgets/data-sources/_index.md:
```diff
@@ -16,8 +16,8 @@ This document describes data sources of data containers.
-* [Microflow](/refguide/microflow-source/) – the data view object is determined by the result of calling the selected microflow. The microflow can take objects in the context as parameter and needs to return a single object.
-* [Nanoflow](/refguide/nanoflow-source/) – objects retrieved are determined by the result of calling the selected nanoflow. The nanoflow can take objects in the context as a parameter and needs to return a single object. 
+* [Microflow](/refguide/microflow-source/) – the data view object is determined by the result of calling the selected microflow. The microflow can take variables or objects in the context as parameter and needs to return a single object.
+* [Nanoflow](/refguide/nanoflow-source/) – objects retrieved are determined by the result of calling the selected nanoflow. The nanoflow can take variables or objects in the context as a parameter and needs to return a single object. 
```
'Primitives or objects' sounds more appropriate than 'variables or objects.'

alihcsumer replied:
✨

alihcsumer commented on content/en/docs/refguide/modeling/pages/data-widgets/data-sources/_index.md:
```diff
@@ -35,8 +35,8 @@ Data grids, template grids, and list views are list widgets. Also, some [pluggab
 {{% /alert %}}
-* [Microflow](/refguide/microflow-source/) – objects retrieved are determined by the result of calling the selected microflow. The microflow can take objects in the context as parameter and needs to return a list of objects.
-* [Nanoflow](/refguide/nanoflow-source/) – objects retrieved are determined by the result of calling the selected nanoflow. The nanoflow can take objects in the context as a parameter and needs to return a list of objects. Nanoflow data source is only available for data views and list views. 
+* [Microflow](/refguide/microflow-source/) – objects retrieved are determined by the result of calling the selected microflow. The microflow can take variables or objects in the context as parameter and needs to return a list of objects.
```
'Primitives or objects' sounds more appropriate than 'variables or objects.'

alihcsumer replied:
✨

alihcsumer commented on content/en/docs/refguide/modeling/pages/data-widgets/data-sources/_index.md:
```diff
@@ -35,8 +35,8 @@ Data grids, template grids, and list views are list widgets. Also, some [pluggab
-* [Microflow](/refguide/microflow-source/) – objects retrieved are determined by the result of calling the selected microflow. The microflow can take objects in the context as parameter and needs to return a list of objects.
-* [Nanoflow](/refguide/nanoflow-source/) – objects retrieved are determined by the result of calling the selected nanoflow. The nanoflow can take objects in the context as a parameter and needs to return a list of objects. Nanoflow data source is only available for data views and list views. 
+* [Microflow](/refguide/microflow-source/) – objects retrieved are determined by the result of calling the selected microflow. The microflow can take variables or objects in the context as parameter and needs to return a list of objects.
+* [Nanoflow](/refguide/nanoflow-source/) – objects retrieved are determined by the result of calling the selected nanoflow. The nanoflow can take variables or objects in the context as a parameter and needs to return a list of objects. Nanoflow data source is only available for data views and list views. 
```
'Primitives or objects' sounds more appropriate than 'variables or objects.'

alihcsumer replied:
✨

alihcsumer commented on content/en/docs/refguide/modeling/pages/data-widgets/data-view.md:
```diff
@@ -100,15 +100,15 @@ The 'Conditionally' option was introduced in Mendix 10.7.0.
 ### Condition
 
-If the editable property is set to **Conditionally**, the widgets inside the data view are made editable only if the object of the data view satisfies the specified criteria.
+If the editable property is set to **Conditionally**, the widgets inside the data view are made editable only if the editability expression satisfies the specified criteria.
```
having the value instead of an expression is an option as well

alihcsumer replied:
✨

alihcsumer commented on content/en/docs/refguide/modeling/pages/input-widgets/_index.md:
```diff
@@ -9,35 +9,46 @@ description: "Widgets which can be added to pages to view and edit attributes of
+* Drop-Down
+* Checkbox
+* Radio Buttons
+* Date Picker
```
Considering the warning messages, should these be mentioned in docs as well?

alihcsumer replied:
✨

alihcsumer commented on static/_includes/refguide/data-source-section-link.md:
```diff
@@ -1,4 +1,4 @@
 
-This selects which attribute will be used within the widget. You can only select attributes which have the correct [data type(s)](/refguide/data-types/).
+This selects which attribute or when applicable, which variable will be used within the widget. You can only select attributes or variables which have the correct [data type(s)](/refguide/data-types/).
```
'values' can be used instead of 'attributes or variables' throughout the document, especially when the context is data type.

alihcsumer replied:
✨

weirdwater committed:
Retract the ability to choose variables from selectors

LEGIO-SEXTA-FERRATA committed:
Review: address review threads

LEGIO-SEXTA-FERRATA committed:
Review: fix editability expression & remove redundant variable using widgets list

ConnorLand committed:
Edit

ConnorLand committed:
Edit

LEGIO-SEXTA-FERRATA committed:
Review: adjust text box picture with a variable usage

LEGIO-SEXTA-FERRATA commented:
Here are the screenshots that need updating:

- The SS in [this section](https://docs.mendix.com/quickstarts/native-mobile-app/#nanoflow-use-case) does not feature the (new) variables button
- The SS at the 4th step of [this section](https://docs.mendix.com/quickstarts/responsive-web-app/#selecting-the-app-template) does not feature the (new) variables button
- The SS in [this section](https://docs.mendix.com/refguide/studio-pro-overview/#introduction) does not feature the (new) variables button
- The **"Attribute"** setting in the dialog in the screenshot in [this section](https://docs.mendix.com/refguide/setting-up-data-validation/#required-validation-on-inputs-reference-selectors-and-drop-downs) is now changed to **"Value"**
- The screenshot in [this section](https://docs.mendix.com/refguide/xpath-source/#xpath-constraints) is not directly related to variables but we don't have the **"XPath"** data source in Mendix anymore
- The first screenshot in [this section](https://docs.mendix.com/refguide/page/#add-elements) does not feature the (new) variables button
- The first screenshot in [this section](https://docs.mendix.com/refguide/page/#structure-mode) does not feature the (new) variables button
- The screenshot in [this section](https://docs.mendix.com/refguide/snippet/#introduction) does not feature the (new) variables button
- The **"Attribute (path)"** setting in [this section](https://docs.mendix.com/refguide/input-reference-set-selector/#properties) is not changed to **"Value"**
- The first screenshot in [this section](https://docs.mendix.com/refguide/page/#add-elements) does not feature the (new) variables button
- I modified [this](https://github.com/mendix/docs/blob/d776ec04440cbe7228c45d9dc4d6bb235e0baf3b/content/en/docs/refguide/modeling/pages/input-widgets/text-box.md?plain=1#L29) screenshot to demonstrate the variables with input widgets; while many other input widgets also are capable of making use of variables, we wanted to use the text box as an example rather than redundantly making the same change for all of the widgets

This is the new variables button:
![variables-button](https://github.com/user-attachments/assets/0fe4d199-5c1f-4657-9fcb-5701b42263b1)

ConnorLand commented:
I'll go back and update screenshots for this PR later this week.
